Form_validation.php

```diff
@@ -0,0 +1,88 @@
+<?php 
+  $valid = array();
+  $errors = array();
+  if ($_POST) {
+
+    $title = $_POST['title'];
+
+    if (!$title) {
+      $errors['title'] = 'Името е задължително поле.';
+      echo $errors['title'];
+      } 
+      elseif (strlen($title) > 150) 
+      {
+        $errors['title'] = 'Името има максимална дължина 150 символа.';
+		echo $errors['title']; 
+      } 
+      else {
+        $valid['title'] = $title;   
+      }
+
+    $teacher = $_POST['teacher'];
+    if (!$teacher) {
+      $errors['teacher'] = 'Името на преподавател е задължително поле.';
+	  echo $errors['teacher'];   
+      } 
+      elseif (strlen($teacher) > 200) 
+      {
+        $errors['teacher'] = 'Името на преподавател има максимална дължина 200 символа.';
+        echo $errors['teacher'];       
+      } 
+      else {
+        $valid['teacher'] = $teacher;   
+      }
+
+      $description = $_POST['description'];
+      if (!$description) {
+        $errors['description'] = 'Описанието на дисциплината е задължително поле.';
+        echo $errors['description'];   
+      } 
+      elseif (strlen($description) < 10) 
+      {
+        $errors['description'] = 'Описанието на дисциплината има минимална дължина 10 символа.';
+        echo $errors['description'];  
+      } 
+      else {
+        $valid['description'] = $description;   
+      }
+
+      $group = $_POST['group'];
+      if(!$group) {
+        //it is filled with the first option by default, so can not be missing
```

Comment:
In such case this is called "dead code" and you may want to delete it because it is unreachable. **BUT:** Also: never trust what comes from the fronted. For example I could use something like: https://www.postman.com/ and combine a POST query against the url where your backend is located by totally bypassing all your frontend validations. This is the reason why we always do validations both on frontend level and backend level as developers and we are basically "repeating" them - due to security reasons. The same goes for DB level. We usually have the same constraints that we have on the frontend and backend on the DB level too. This for example becomes useful when there are 2 types of backends connected to the same DB, and imagine the possibilities - the second backend may be missing some / all validations at this case the DB constraints come to play to protect the integrity of data. Also very often after an application that is in production there could happen some inconsistency on DB level (for example because of some bug) and the whole application to become unusable. In order to fix such situations developers need to manually execute SQL scripts against the DB. And in this case the DB level constraints are protecting us once again from ourselves and keeping data integrity.
**tl;dr;** please add the validation here as it actually CAN be missing.